cli/pcluster/config/cfn_param_types.py

```diff
@@ -55,7 +55,14 @@ def from_cfn_params(self, cfn_params):
         cfn_converter = self.definition.get("cfn_param_mapping", None)
         if cfn_params:
             cfn_value = get_cfn_param(cfn_params, cfn_converter) if cfn_converter else "NONE"
-            self.value = self.get_value_from_string(cfn_value)
+            self.value = (
+                None
+                if self.key == "cluster_resource_bucket"
+                and get_cfn_param(cfn_params, "RemoveBucketOnDeletion") == "True"
```

Comment:
Minor: is there a relatively cleaner way to write this logic? For example we can extract this logic into a function `handle_special_param` or something?

Reply:
For creating a function `handle_special_param`, different special parameters will have different logic for this `if` and `else` here. I am not sure how to extract this logic.
I am thinking about another way: have another param class for 'cluster_resource_bucket', inherited from CfnParam, rewrite the method `def from_cfn_params`, but it is also not a clean way